### PR TITLE
Feature/sms documentation

### PR DESCRIPTION
**Pull request by HannesBenson**
## Change description

This PR:
- Updates the link to the Slack documentation
- Adds Slack & SMS to the channels page
- Remove SMS from the coming soon bit
- Add `phoneNumbers` to the graphql users' section
- Add `phone_numbers` to the users' REST api section.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

[MAG-177](https://linear.app/magicbell/issue/MAG-177/update-the-documentation-regarding-user-creationupdating-to-specify)
[MAG-178](https://linear.app/magicbell/issue/MAG-178/update-the-documentation-to-highlight-the-fact-that-we-now-have-an-sms)

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: docs/channels.mdx [hunk 1]
```diff
@@ -63,3 +64,13 @@ Push notifications are often a crucial component of mobile applications.
 
 Push notifications for the web are a great way to reach users working on desktop
 machines instantly. A good reminder is users must enable desktop notifications.
+
+## Slack
+
+Reach your users with Slack notifications powered by MagicBell. MagicBell supports
+both webhook and oauth Slack apps to allow you all the control you need.
```
Comment (HannesBenson) [inline docs/channels.mdx:71]: Can I get some help with the text here?

File: docs/channels.mdx [hunk 2]
```diff
@@ -63,3 +64,13 @@ Push notifications are often a crucial component of mobile applications.
...
+
+## SMS
+
+SMS notifications helps you deliver notifications to all your users, even those
+in places with no internet.
```
Comment (HannesBenson) [inline docs/channels.mdx:76]: Can I get some help with the text here?

Comment (unamashana) [reply docs/channels.mdx:76]: @HannesBenson Looks good. I'd probably add a bit of humor. Or those with internet that still like a little old fashioned touch :)

File: docs/channels.mdx (hunk 1 above)
Comment (unamashana) [reply docs/channels.mdx:71]: @HannesBenson looks pretty good to me!

**Review by unamashana — APPROVED** [review]

**Pull request merged**